okapi-core/src/test/java/org/folio/okapi/PullTest.java

```diff
@@ -115,6 +118,13 @@ private void td(TestContext context, Async async) {
       });
       return;
     }
+    if (otherServer != null) {
+      otherServer.close().onComplete(res -> {
+        otherServer = null;
+        td(context, async);
+      });
+      return;
+    }
     vertx.close(context.asyncAssertSuccess());
     async.complete();
   }
```

Comment:
[Vertx.close](https://vertx.io/docs/apidocs/io/vertx/core/Vertx.html#close--) automatically closes all resources (Verticles, servers, ...) opened by it, no need to do it manually.
Do you have a use case or reproducer where this doesn't work? If yes this is a bug in Vertx and should be reported upstream.

    @After
    public void tearDown(TestContext context) {
      vertx.close().onComplete(context.asyncAssertSuccess());
    }

or

    @After
    public void tearDown(TestContext context) {
      vertx.close(context.asyncAssertSuccess());
    }

Reply:
ok.. I was probably being paranoid.. I'll test a little on my slow Windows (because in Vmware player) .